### ISSUE TITLE
External Fetcher - DELE without RETR

### ISSUE THREAD

**Issue by sprintman** [hmailserver/hmailserver#253]
I am getting errors with the external fetcher when there is no new messages to be downloaded but there is an old message to be deleted. After the error, the old message remains on the external server and hMailServer seems to ignore it in future sessions, neither fetching nor deleting it. I never saw this issue with 5.6.7-B2425 or earlier.
[https://www.hmailserver.com/forum/viewtopic.php?f=10&t=32493](https://www.hmailserver.com/forum/viewtopic.php?f=10&t=32493)
```
"DEBUG"	6816	"2018-03-07 18:11:08.793"	"Adding task ExternalFetchTask to work queue External fetch queue"
"DEBUG"	4260	"2018-03-07 18:11:08.793"	"Executing task ExternalFetchTask in work queue External fetch queue"
"DEBUG"	4260	"2018-03-07 18:11:08.793"	"Retrieving messages from external account mail"
"DEBUG"	4260	"2018-03-07 18:11:08.794"	"Pre-creating session 1629"
"TCPIP"	4260	"2018-03-07 18:11:08.794"	"Connecting to xxx.xxx.xxx.xxx:995..."
"DEBUG"	6664	"2018-03-07 18:11:08.800"	"TCP connection started for session 1629"
"DEBUG"	6664	"2018-03-07 18:11:08.800"	"Performing SSL/TLS handshake for session 1629. Verify certificate: True, Expected remote host name: mail.domain.com"
"DEBUG"	6656	"2018-03-07 18:11:08.816"	"Certificate verification succeeded for session 1629."
"TCPIP"	6664	"2018-03-07 18:11:08.829"	"TCPConnection - TLS/SSL handshake completed. Session Id: 1629, Remote IP: xxx.xxx.xxx.xxx, Version: TLSv1.2, Cipher: ECDHE-RSA-AES128-GCM-SHA256, Bits: 128"
"POP3C"	6664	1629	"2018-03-07 18:11:09.029"	"xxx.xxx.xxx.xxx"	"RECEIVED: +OK Dovecot ready."
"POP3C"	6664	1629	"2018-03-07 18:11:09.030"	"xxx.xxx.xxx.xxx"	"SENT: USER mail@domain.com"
"POP3C"	6656	1629	"2018-03-07 18:11:09.035"	"xxx.xxx.xxx.xxx"	"RECEIVED: +OK"
"POP3C"	6656	1629	"2018-03-07 18:11:09.035"	"xxx.xxx.xxx.xxx"	"SENT: ***"
"POP3C"	6664	1629	"2018-03-07 18:11:09.142"	"xxx.xxx.xxx.xxx"	"RECEIVED: +OK Logged in."
"POP3C"	6664	1629	"2018-03-07 18:11:09.142"	"xxx.xxx.xxx.xxx"	"SENT: UIDL"
"POP3C"	6656	1629	"2018-03-07 18:11:09.158"	"xxx.xxx.xxx.xxx"	"SENT: DELE 51"
"POP3C"	6656	1629	"2018-03-07 18:11:09.164"	"xxx.xxx.xxx.xxx"	"RECEIVED: +OK Marked to be deleted."
"ERROR"	6656	"2018-03-07 18:11:09.164"	"Severity: 3 (Medium), Code: HM5136, Source: TCPConnection::AsyncReadCompleted, Description: An error occured while parsing data. Data size: 0 Remote IP: xxx.xxx.xxx.xxx"
"DEBUG"	6656	"2018-03-07 18:11:09.165"	"Ending session 1629"
"DEBUG"	4260	"2018-03-07 18:11:09.165"	"Completed retrieval of messages from external account."
"ERROR"	6656	"2018-03-07 18:11:09.165"	"Severity: 2 (High), Code: HM4208, Source: ExceptionHandler::Run, Description: An error occured while executing 'IOCPQueueWorkerTask'"
"DEBUG"	6656	"2018-03-07 18:11:09.165"	"Logging exception.."
"ERROR"	6656	"2018-03-07 18:11:09.278"	"Severity: 1 (Critical), Code: HM5519, Source: StackLogger::Log, Description: An error has been detected. A mini dump has been written to C:\hMailServer\Logs\minidump_2018-03-07 181109_{0F93C8D0-6197-4E1E-A4B1-78EBB49D585F}.dmp"
"DEBUG"	6656	"2018-03-07 18:11:09.278"	"Completed logging of exception..."
```

**Comment by sprintman** [comment]
Reverting 000131a35a650004a86a33663b07c4d648431bf3 fixes this

**Comment by derelvis** [comment]
I would like to support this request. Having this issue for a long time now without solution.
Asked for help in the forum and tried other/new versions but no success. 
[https://www.hmailserver.com/forum/viewtopic.php?f=10&t=33837](https://www.hmailserver.com/forum/viewtopic.php?f=10&t=33837)
Maybe it is a simple fix or just reverting changed code back as mentioned in the pull request. But I am not able to do this myself for testing :(
Would be very nice if you could have a quick look into it, @martinknafve 
Many many thanks in advance and of course for all the work you have done before! It is a great project and I respect your time spending on this very much.
Daniel

**Comment by martinknafve** [comment]
This fix has now been included in 5.7. Closing.
